template/registry/number.go

```diff
@@ -0,0 +1,231 @@
+package registry
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"golang.org/x/text/currency"
+)
+
+// https://github.com/unicode-org/message-format-wg/blob/122e64c2482b54b6eff4563120915e0f86de8e4d/spec/registry.xml#L147
+
+var numberRegistryF = &Func{
+	Name:           "number",
+	Description:    "Locale-sensitive number formatting",
+	F:              numberF,
+	MatchSignature: nil, // Not allowed to use in matching context
+	FormatSignature: &Signature{
+		Input: true,
+		ValidateInput: func(a any) error {
+			if _, err := castAs[float64](a); err != nil {
+				return fmt.Errorf("unsupported type: %T: %w", a, err)
+			}
+
+			return nil
+		},
+		Options: Options{
+			{
+				Name:           "compactDisplay",
+				Description:    `Only used when notation is "compact".`,
+				PossibleValues: []any{"short", "long"},
+				Default:        "short",
+			},
+			{
+				Name: "currency",
+				Description: `The currency to use in currency formatting.
+Possible values are the ISO 4217 currency codes, such as "USD" for the US dollar,
+"EUR" for the euro, or "CNY" for the Chinese RMB — see the
+Current currency &amp; funds code list
+(https://www.unicode.org/cldr/charts/latest/supplemental/detailed_territory_currency_information.html).
+There is no default value; if the style is "currency", the currency property must be provided.`,
+				ValidateValue: func(a any) error {
+					_, err := castAs[currency.Unit](a)
```

Comment:
This does not validate currency value, right?

Reply:
in this context: 
only `currency.Unit` can be passed as option value

Reply:
so no casting

Reply:
i mean, there is nothing to validate, if argument is currency.Unit, no ? 🤔

Reply:
i think alternative is better
```go
if _, ok := a.(currency.Unit); !ok {
return ...
}

return nil
```

Reply:
Mostly it seems to be ok, `currency.Unit` does not expose its value -  https://cs.opensource.google/go/x/text/+/refs/tags/v0.14.0:currency/currency.go;l=69

Effectively, consumers of the currency package cannot change the index directly.

However, zero value `currency.Unit` might cause an issue.

Reply:
yeah, it can be a problem